### DISCUSSION TITLE
Request for Improved Validation Speed with TypeBox in FETS Library

### DISCUSSION THREAD

**Discussion by MineMon2** [ardatan/feTS#398] [Ideas]
**Is your feature request related to a problem? Please describe.**
Issue Description:
As an enthusiastic user of the FETS library, I appreciate its potential and the fact that it leverages JSON Schema, making it compatible with OPEN API. However, during my usage, I've noticed that the current validation process, powered by ZOD, tends to be slower when dealing with large datasets. After some research, I came across TypeBox, a similar library known for its significantly faster execution speed.

Feature Request:
I kindly request the development team to consider integrating TypeBox as an alternative validation option within the FETS library. By doing so, users can benefit from improved validation performance, especially in scenarios where speed is crucial.

Benefits of TypeBox:
TypeBox has gained popularity due to its efficient validation capabilities, making it an ideal candidate to enhance the FETS library's overall user experience. By adopting TypeBox, we can take advantage of its speed and performance benefits while maintaining JSON Schema compatibility.

Community Impact:
Introducing TypeBox into the FETS library would not only benefit current users by accelerating validation but also attract new developers seeking a performant JSON Schema-based solution. This addition could potentially increase the library's adoption and foster a more vibrant and engaged user community.

Implementation Considerations:
While considering the incorporation of TypeBox, it would be prudent to conduct thorough compatibility testing to ensure seamless integration with the existing functionalities of the FETS library. Additionally, providing clear documentation and examples on how to leverage TypeBox for validation within the FETS ecosystem would be greatly appreciated by users.

Thank you for considering this feature request. Your continuous efforts in improving the FETS library are commendable, and I believe that adopting TypeBox will elevate its performance and broaden its appeal to developers worldwide.

Best regards,

https://moltar.github.io/typescript-runtime-type-benchmarks/

**Comment by ardatan** [comment]
Thanks for the feedback! Now feTS supports TypeBox instead of Zod!
